client/lens.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package client
+
+import (
+	"github.com/lens-vm/lens/host-go/config/model"
+	"github.com/sourcenetwork/defradb/datastore"
+	"github.com/sourcenetwork/immutable/enumerable"
+)
+
+type LensConfig struct {
+	SourceSchema      string
+	DestinationSchema string
+	model.Lens
+}
+
+type LensRegistry interface {
+	RegisterLens(datastore.Txn, LensConfig) error
+	MigrateUp(enumerable.Enumerable[map[string]any], string) (enumerable.Enumerable[map[string]any], error)
+	MigrateDown(enumerable.Enumerable[map[string]any], string) (enumerable.Enumerable[map[string]any], error)
```

Comment:
Discussed over Discord a bit a while ago, but noting here for reference etc, at the time of comment the PR was not ready for even draft review, only as a quick reference in case useful for other PRs.  Will of course be much better documented by the time ready for review.